stackslib/src/net/p2p.rs

```diff
@@ -4377,6 +4390,12 @@ impl PeerNetwork {
             self.stacks_tip.is_nakamoto
         };
 
+        // TODO: Test this!
+        let stacks_tip_cbh = NakamotoChainState::get_coinbase_height(
+            &mut chainstate.index_conn(),
+            &new_stacks_tip_block_id,
+        )?;
```

Comment:
If this is `None`, then you'd need to confirm that the Stacks tip is a 2.x block, and then set the coinbase height to that of the Stacks 2.x block's height.